### PR TITLE
fix build sbt to comply with intellij expectations

### PR DESCRIPTION
**Pull request by Gryfit**
I wanted to pick up the Keras PR on @Avasil request, but before that I have to fix intellij issues with this project.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -4,6 +4,16 @@ organization in ThisBuild := "me.shadaj"
 
 scalaVersion in ThisBuild := "2.12.8"
 
+Compile / scalaSource := baseDirectory.value / "src" /  "main" / "scala"
```
Comment (PawelLipski) [inline build.sbt:7]: While you're here... pls also provide the list of required software/C headers (`re2` etc.) to install before `sbt test:compile` in README.md

Comment (Gryfit) [reply build.sbt:7]: I suppose ensuring setup for scala native (including optional libraries)  is obligatory, I don't know which exactly but it's safe to say all
```
And following runtime library dependencies:
libunwind 0.99 or newer (built-in on macOS)
Boehm GC 7.6.0 (optional)
Re2 2017-01-01 (optional)
zlib 1.2.8 or newer (optional)
```
From https://scala-native.readthedocs.io/en/v0.3.9-docs/user/setup.html

Comment (PawelLipski) [reply build.sbt:7]: Ok! pls add to README so that it's not lost

File: build.sbt [hunk 2]
```diff
@@ -4,6 +4,16 @@ organization in ThisBuild := "me.shadaj"
...
+
+Test / scalaSource := baseDirectory.value / "src" / "test" / "scala"
+
+// dependencies for scalapy-tensorflow-source
+libraryDependencies += "me.shadaj" %%% "scalapy-core" % "0.3.0"
```
Comment (PawelLipski) [inline build.sbt:12]: Actually... how come could this project ever compile before without these deps? :hushed: or maybe: why are they only required by IntelliJ (at least that's how I understand the PR title), but not by a standalone sbt?

File: build.sbt [hunk 3]
```diff
@@ -4,6 +4,16 @@ organization in ThisBuild := "me.shadaj"
...
+libraryDependencies += "me.shadaj" %%% "scalapy-numpy" % "0.1.0+3-046d1d67"
+libraryDependencies += "org.scalatest" %%% "scalatest" % "3.1.0-SNAP8" % Test
```
Comment (PawelLipski) [inline build.sbt:14]: Hmm why use snapshot release? A regular scalatest 3.1.0 is already released

File: build.sbt [hunk 4]
```diff
@@ -4,6 +4,16 @@ organization in ThisBuild := "me.shadaj"
 
 scalaVersion in ThisBuild := "2.12.8"
 
+Compile / scalaSource := baseDirectory.value / "src" /  "main" / "scala"
+
+Test / scalaSource := baseDirectory.value / "src" / "test" / "scala"
```
Comment (PawelLipski) [inline build.sbt:9]: Hmm weird that IntelliJ doesn't infer those Scala source locations on its own :thinking:

File: build.sbt (hunk 1 above)
Comment (PawelLipski) [reply build.sbt:7]: (at least the link to scala-native setup)

Comment (Gryfit) [reply build.sbt:7]: ok

File: build.sbt (hunk 2 above)
Comment (Gryfit) [reply build.sbt:12]: Ok sooo this project is complicated.
If you look at the project structure you will see scalapy-tensorflow-soruces without dependecies and then JVM and Native versions with dependencies. Look below on `scalaPyTensorFlowCross`.
So we can say that all deps are production deps and there are no dev deps so intellij is blind.

File: build.sbt (hunk 3 above)
Comment (Gryfit) [reply build.sbt:14]: well those are production dependencies of this release, copied form `scalaPyTensorFlowCross`

File: build.sbt (hunk 4 above)
Comment (Gryfit) [reply build.sbt:9]: Maybe because of the insane project in project structure. Anyway pointing them by hand solves the issue.

Comment (shadaj) [reply build.sbt:9]: Recent versions of IntelliJ are supposed to support Scala.js / Scala Native project structures. What's the type of errors you are facing?

File: build.sbt (hunk 2 above)
Comment (shadaj) [reply build.sbt:12]: This shouldn't be necessary, the dependency is brought in transitively by the dependency on `scalapy-numpy`.

File: build.sbt (hunk 3 above)
Comment (shadaj) [reply build.sbt:14]: Can we upgrade to `3.1.2`?

**Review by shadaj — CHANGES_REQUESTED** [review]

**Pull request closed without merging**